R/checks.R

```diff
@@ -1,101 +1,107 @@
 
+  invisible(NULL)
+}
+
+check_value_order <- function(low, high, target = NULL, call) {
+  check_number_decimal(low, call = call)
+  check_number_decimal(high, call = call)
+  check_number_decimal(target, allow_null = TRUE, call = call)
+
   if (!is.null(target)) {
-    if (length(target) != 1 || !is.numeric(target) || is.na(target)) {
-      rlang::abort("'target' should be a single numeric value.")
-    }
     ord <- low < target & target < high
     if (!ord) {
-      rlang::abort("The values should be `low < target < high`.")
-    }
-  } else {
-    ord <- low < high
-    if (!ord) {
-      rlang::abort("The values should be `low < high`.")
+      cli::cli_abort("The values should be {.code low < target < high}  (actual
+                     are {low}, {target}, and {high}).", call = call)
     }
   }
+
+  ord <- low < high
+  if (!ord) {
+    cli::cli_abort("The values should be {.code low < high} (actual are {low}
+                   and {high}).", call = call)
+  }
+
   invisible(NULL)
 }
 
-is_vector_args <- function(values, d) {
+is_vector_args <- function(values, d, call) {
   if (!is.vector(values) || !is.numeric(values)) {
-    rlang::abort("'values' should be a numeric vector.")
+    cli::cli_abort("{.arg values} should be a numeric vector.", call = call)
   }
   if (!is.vector(d) || !is.numeric(d)) {
-    rlang::abort("'d' should be a numeric vector.")
+    cli::cli_abort("'d' should be a numeric vector.", call = call)
   }
   if (length(values) != length(d)) {
-    rlang::abort("'values' and 'd' should be the same length.")
+    cli::cli_abort("'{.arg values}' and '{.arg d}' should be the same length.",
+                   call = call)
   }
   invisible(TRUE)
 }
 
 
-check_args <- function(arg, x, use_data, fn, type = "low") {
+check_args <- function(arg, x, use_data, fn, type = "low", call) {
   if (rlang::is_missing(arg)) {
     if (use_data) {
-      type <- rlang::arg_match0(type, c("low", "high", "target"))
+      type <- rlang::arg_match0(type, c("low", "high", "target"), error_call = call)
       .fn <- switch(type, low = min, high = max, target = stats::median)
       arg <- .fn(x, na.rm = TRUE)
     } else {
-      rlang::abort(
-        glue::glue("In `{fn}()`, argument '{type}' is required when 'new_data = FALSE'.")
-      )
+      cli::cli_abort("In {.fn {fn}}, argument {.arg {type}} is required when
+                     {.code new_data = FALSE}.", call = call)
     }
   }
   arg
 }
 
-check_scale <- function(x) {
-  if (length(x) != 1 || !is.numeric(x) || is.na(x)) {
-    rlang::abort("The scale parameter should be a single numeric value.")
-  }
-  if (x <= 0) {
-    rlang::abort("The scale parameter great then zero.")
-  }
-
+check_scale <- function(x, arg, call) {
+  check_number_decimal(x, min = 0, arg = arg, call = call)
   invisible(NULL)
 }
 
-is_d_input <- function(x) {
-  tmp <- purrr::map(x, check_numeric, input = "desirability")
-  tmp <- purrr::map(x, check_unit_range)
+is_d_input <- function(x, call) {
```

Comment:
The next PR will use a constructor to check these things and give the desirability vectors classes `c("numeric", "desirability")` without going full vctrs. 

That will make these checks a lot easier/better. The current method would let something like `roc_auc` in without an error.